beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java

```diff
@@ -246,10 +253,7 @@ private SafeFuture<Void> setExecutionData(
     // Post-Deneb: Execution Payload / Execution Payload Header + KZG Commitments
     final ExecutionPayloadResult executionPayloadResult =
         executionLayerBlockProductionManager.initiateBlockAndBlobsProduction(
-            // kzg commitments are supported: we should have already merged by now, so we
-            // can safely assume we have an executionPayloadContext
-            executionPayloadContext.orElseThrow(
-                () -> new IllegalStateException("Cannot provide kzg commitments before The Merge")),
+            executionPayloadContext.orElseThrow(),
```

Comment:
We can just .get() here. It's safe if we check only the emptiness above

Reply:
code style barks at me